dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -53,7 +57,14 @@ public DBI build(Environment environment,
         final String validationQuery = configuration.getValidationQuery();
```

Comment:
Interesting note.  The JDBC driver could (and should) be configured (e.g., [socket settings](https://jdbc.postgresql.org/documentation/head/connect.html) for the Postgres driver) to alleviate the failure scenario for the Tomcat pool, but the HikariCP pool sounds like a nice piece of work.